### PR TITLE
Add better react and react-native compatibility

### PR DESCRIPTION
**Pull request by ragibkl**
Hi maintainers,

I'm trying to integrate this library into my react and react-native projects.
I believe, the following small changes will help:
- add `buffer` dependency, and polyfill global `Buffer`
- add `process` dependency, and minimally polyfill global `process`
- add `browserify-zlib` dependency for browser type environment
- add `react-zlib-js` dependency for react-native type environment

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,15 +27,19 @@
...
     "base64url": "^3.0.1",
+    "buffer": "^5.5.0",
     "es6-promise": "^4.2.8",
     "lodash": "^4.17.15",
     "long": "^4.0.0",
     "node-forge": "^0.8.5",
-    "uuid": "^3.3.3"
+    "process": "^0.11.10",
+    "uuid": "^3.3.3",
+    "zlib-browserify": "^0.0.3"
```
Comment (ragibkl) [inline package.json]: I might have made a mistake here, this should be `browserify-zlib`

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "node-jose",
-  "version": "1.1.3",
```
Comment (linuxwolf) [inline package.json]: We like to keep the version increments as independent commits/PRs.  Please revert this line for now.

File: lib/index.js [hunk 3]
```diff
@@ -9,6 +9,20 @@ if (typeof Promise === "undefined") {
   require("es6-promise").polyfill();
 }
 
+if (typeof Buffer === "undefined") {
+  const g = global || window;
+  g.Buffer = require("buffer/").Buffer;
```
Comment (linuxwolf) [inline lib/index.js]: Did you find the trailing "/" necessary?  It shouldn't be, and I've not run into problems elsewhere, but I haven't tried integrating with `react-native` either.

File: lib/index.js [hunk 4]
```diff
@@ -9,6 +9,20 @@ if (typeof Promise === "undefined") {
...
+}
+
+if (typeof process === "undefined") {
+  const g = global || window;
+  g.process = require("process/");
```
Comment (linuxwolf) [inline lib/index.js]: same as line 14 above.

File: lib/index.js (hunk 3 above)
Comment (ragibkl) [reply lib/index.js]: I tested this a bit more, and it seems you are right. The trailing `/` is not necessary here.

I tested with node, react and react-native and it seems to work just fine. I think I can remove it here.

File: lib/index.js (hunk 4 above)
Comment (ragibkl) [reply lib/index.js]: I tested this a bit more, and you are right. The trailing slash can be removed. I'll make the change.

File: package.json (hunk 2 above)
Comment (ragibkl) [reply package.json]: I've updated this part as requested.

File: lib/index.js [hunk 5]
```diff
@@ -9,6 +9,20 @@ if (typeof Promise === "undefined") {
   require("es6-promise").polyfill();
 }
 
+if (typeof Buffer === "undefined") {
+  const g = global || window;
+  g.Buffer = require("buffer").Buffer;
```
Comment (ragibkl) [inline lib/index.js]: @linuxwolf , I've noticed that you don't use `const` variable declaration anywhere in the project. For consistency, I can change these lines to either the following:
```
var g = global || window;
g.Buffer = require("buffer").Buffer;
```
OR
```
(global || window).Buffer = require("buffer").Buffer;
```

Comment (linuxwolf) [reply lib/index.js]: Yeah, that's because this project started quite a few years ago, before `const` and `let` were as universally supported as they are now.

Thanks for sticking to consistency.

Comment (linuxwolf) [reply lib/index.js]: I'm personally fine with either.

Comment (ragibkl) [reply lib/index.js]: I've updated to 2nd option for consistency. Marking this as resolved.

**Comment by ragibkl** [comment]
Hi @linuxwolf ,

I don't have any other changes for this PR for now. However, after more testing in `react-native`, I realised that I was getting invalid results.

I would like to help address that separately, and have created another [PR](https://github.com/cisco/node-jose/pull/285) for that.
